src/ncp/ncp_base_dispatcher.cpp

```diff
@@ -351,10 +351,12 @@ NcpBase::PropertyHandler NcpBase::FindGetPropertyHandler(spinel_prop_key_t aKey)
         OT_NCP_GET_HANDLER_ENTRY(SPINEL_PROP_CHANNEL_MANAGER_DELAY),
         OT_NCP_GET_HANDLER_ENTRY(SPINEL_PROP_CHANNEL_MANAGER_SUPPORTED_CHANNELS),
         OT_NCP_GET_HANDLER_ENTRY(SPINEL_PROP_CHANNEL_MANAGER_FAVORED_CHANNELS),
+#if OPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE
```

Comment:
I think only restricting `SPINEL_PROP_CHANNEL_MANAGER_CHANNEL_SELECT` (and its Get/Set handler in ncp_base_ftd.cpp) is enough. I've tried building it and it can work:
```
./script/cmake-build simulation -DOT_CHANNEL_MONITOR=OFF  
```

But you need to add another small change in `ChannelManager::SetAutoNetworkChannelSelectionEnabled`: Add a `OT_UNUSED_VARIABLE` for `aEnabled`.

Reply:
Oh just saw your discussion in the issue. Make sense to me. Please ignore the above comments.